pkg/virt-launcher/virtwrap/nichotplug.go

```diff
@@ -95,6 +98,43 @@ func (vim *virtIOInterfaceManager) hotplugVirtioInterface(vmi *v1.VirtualMachine
 	return nil
 }
 
+func (vim *virtIOInterfaceManager) syncInterfacesLinkState(currentDomain *api.Domain, desiredDomain *api.Domain) error {
+
+	currentDomainIfacesByAlias := indexedDomainInterfaces(currentDomain)
+	for _, desiredIface := range desiredDomain.Spec.Devices.Interfaces {
+		curIface, ok := currentDomainIfacesByAlias[desiredIface.Alias.GetName()]
+		if !ok {
+			return fmt.Errorf("could not find requested domain interface %s in domain", desiredIface.Alias.GetName())
+		}
+		if err := vim.updateDomainWithLinkState(&curIface, requiredLinkAction(curIface, desiredIface)); err != nil {
+			return err
+		}
```

Comment:
How about:
If the desired state is different from the current state -> set the current state to the desired state.

Reply:
We could do something along the lines of:
```go
func isLinkStateEqual(iface1, iface2 api.Interface) bool {
	if iface1.LinkState == nil && iface2.LinkState == nil {
		return true
	}

	if iface1.LinkState == nil || iface2.LinkState == nil {
		return false
	}

	return iface1.LinkState.State == iface2.LinkState.State
}
```

```go
if !isLinkStateEqual(desiredIface, curIface) {
	currentIface.LinkState = desiredIface.LinkState
	vim.updateInterface(currentIface)
}
```

Doing so will allow us to be indifferent to the difference between the desired and current states.